forloop_modules/utils/url_template_builder.py

```diff
@@ -0,0 +1,117 @@
+import difflib
+from urllib.parse import urlparse
+
+
+class UrlTemplateBuilder:
+    """
+    A class for constructing URL templates and formatting them with arguments.
+
+    Args:
+    ----
+        url_1 (str): The first URL.
+        url_2 (str): The second URL.
+
+    Attributes:
+    ----------
+        templated_count (int): The count of templated arguments in the URL template.
+        template_args (set): A set of argument names used in the URL template.
+        url_template (str): The constructed URL template as a template string with `template_args` as placeholders.
+
+    Methods:
+    -------
+        format_url: Format the URL template with the given arguments.
+    """
+
+    def __init__(self, url_1: str, url_2: str) -> None:
+        self._url_1 = urlparse(url_1, scheme='http')
+        self._url_2 = urlparse(url_2, scheme='http')
+
+        self.templated_count = 0
+        self.template_args = set()
+        self.url_template = ''
+        self._construct_url_template()
+
+    def format_url(self, **kwargs) -> str:
+        """Format the URL template with the given arguments."""
+        if any(arg not in self.template_args for arg in kwargs):
+            raise ValueError('One of the provided arguments is not the template argument')
+        return self.url_template.format(**kwargs)
+
+    def _construct_url_template(self) -> str:
+        if self._url_1.netloc != self._url_2.netloc:
+            raise ValueError('The URLs do not have the same netloc')
+
+        path_template = self._build_path_template()
+        query_template = self._build_query_template()
+
+        self.url_template = f'{self._url_1.scheme}://{self._url_1.netloc}{path_template}?{query_template}'
+
+        return self.url_template
+
+    def _build_path_template(self) -> str:
+        paths_1 = self._url_1.path.split('/')
+        paths_2 = self._url_2.path.split('/')
+
+        template_paths = []
+        for arg_1, arg_2 in zip(paths_1, paths_2):
+            if arg_1 == arg_2:
+                template_paths.append(arg_1)
+                continue
```

Comment:
I recommend not using "continue", in this case it means rearranging the inner parts of the loop